### PR TITLE
[Merged by Bors] - chore(RingTheory/{Algebraic, Localization/Integral}): rename decls to use dot notation

### PR DESCRIPTION
**Pull request by adomani**
This PR tests a string-based tool for renaming declarations.

Inspired by [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Renaming.20decls.20using.20LSP/near/402399853), I am trying to reduce the diff of #8406.

This PR makes the following renames:

| From                                        | To
|---------------------------------------------|------------------------------------------
| `FG_adjoin_singleton_of_integral`           | `IsIntegral.fg_adjoin_singleton`
| `FG_adjoin_of_finite`                       | `fg_adjoin_of_finite`
| `isIntegral_quotient_of_isIntegral`         | `Algebra.IsIntegral.quotient`
| `isAlgebraic_of_larger_base_of_injective`   | `IsAlgebraic.of_larger_base_of_injective`
| `isAlgebraic_of_pow`                        | `IsAlgebraic.of_pow`
| `isAlgebraic_algHom_of_isAlgebraic`         | `IsAlgebraic.algHom`
| `isAlgebraic_algebraMap_of_isAlgebraic`     | `IsAlgebraic.algebraMap`
| `isAlgebraic_of_finite`                     | `IsAlgebraic.of_finite`
| `isAlgebraic_of_larger_base`                | `IsAlgebraic.of_larger_base`
| `isIntegral_add`                            | `IsIntegral.add`
| `isIntegral_map_of_comp_eq_of_isIntegral`   | `IsIntegral.map_of_comp_eq`
| `isIntegral_mul`                            | `IsIntegral.mul`
| `isIntegral_neg`                            | `IsIntegral.neg`
| `isIntegral_ofSubring`                      | `IsIntegral.of_subring`
| `isIntegral_of_finite`                      | `IsIntegral.of_finite`
| `isIntegral_of_isIntegral_mul_unit`         | `IsIntegral.of_mul_unit`
| `isIntegral_of_mem_closure`                 | `IsIntegral.of_mem_closure`
| `isIntegral_of_mem_of_FG`                   | `IsIntegral.of_mem_of_fg`
| `isIntegral_of_pow`                         | `IsIntegral.of_pow`
| `isIntegral_smul`                           | `IsIntegral.smul`
| `isIntegral_sub`                            | `IsIntegral.sub`
| `isIntegral_tower_bot_of_isIntegral`        | `IsIntegral.tower_bot`
| `isIntegral_tower_bot_of_isIntegral_field`  | `IsIntegral.tower_bot_of_field`
| `map_isIntegral`                            | `IsIntegral.map`
| `mem_integralClosure_iff_mem_FG`            | `mem_integralClosure_iff_mem_fg`

---

Affected files:
```bash
FieldTheory/AbelRuffini
FieldTheory/Adjoin
FieldTheory/IsAlgClosed/AlgebraicClosure
FieldTheory/Minpoly/Field
FieldTheory/Normal
FieldTheory/Perfect
FieldTheory/PrimitiveElement
FieldTheory/Separable
FieldTheory/SplittingField/IsSplittingField
NumberTheory/Cyclotomic/Basic
NumberTheory/Cyclotomic/Discriminant
NumberTheory/Cyclotomic/PrimitiveRoots
NumberTheory/Cyclotomic/Rat
NumberTheory/NumberField/Norm
RingTheory/Adjoin/Field
RingTheory/Adjoin/PowerBasis
RingTheory/Algebraic
RingTheory/DedekindDomain/Ideal
RingTheory/DedekindDomain/IntegralClosure
RingTheory/Discriminant
RingTheory/FractionalIdeal
RingTheory/Ideal/Over
RingTheory/IntegralClosure
RingTheory/IntegrallyClosed
RingTheory/Jacobson
RingTheory/Localization/Integral
RingTheory/Norm
RingTheory/Polynomial/Eisenstein/IsIntegral
RingTheory/RingHom/Integral
RingTheory/Trace
```

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Algebraic.lean [hunk 1]
```diff
@@ -136,23 +136,23 @@ theorem isAlgebraic_of_mem_rootSet {R : Type u} {A : Type v} [Field R] [Field A]
 
 open IsScalarTower
 
-theorem isAlgebraic_algebraMap_of_isAlgebraic {a : S} :
+protected theorem IsAlgebraic.algebraMap {a : S} :
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Algebraic.lean:139]: I assume the `protected` was not handled by the tool?

**Review by eric-wieser — APPROVED** [review]
bors d+

Please adjust the PR title to explain the change itself, not why you made it

**Comment by eric-wieser** [comment]
Though perhaps we should address @jcommelin's comment first:

> I want to preface my comment by saying that this is my personal opinion. As a community, we're still very much trying to figure out what the best strategy/policy is, I guess.
> 
> Anyways, I think I would advocate keeping the old names around, and tagging them with `@[deprecated fooBar]` as a pointer to the new name. In a future PR (can be hours/days/weeks later) we remove all the deprecated decls. The big upshot is that there will be a formal trace in the git history that traces the renaming. This will help:
> 
> * people bumping mathlib in their personal projects
> * changelog
> * future automated refactoring tools that do mathlib bumps for us

**Comment by erdOne** [comment]
I think the list of renames should go into the pr description.

**Comment by jcommelin** [comment]
Done

**Comment by eric-wieser** [comment]
@jcommelin, my personal opinion is that using `deprecated` in this way is too large a burden to for mathlib adopt right now, and is not a particularly practical way to encode things into the git history if we want automated tools to read them. Are you ok with forgoing that in this PR?

File: Mathlib/RingTheory/Algebraic.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/RingTheory/Algebraic.lean:139]: Correct, although there is the possibility of flagging declarations that you want to protect in a future version of the renaming tool: the `ilean` files that I use contains where the declaration is defined.

Comment (adomani) [reply Mathlib/RingTheory/Algebraic.lean:139]: However, more basic features are missing in the tool, so I haven't given much thought to this.

**Pull request closed without merging**